src/stores/NearbyLocationsStore.js

```diff
@@ -1,58 +1,53 @@
 // @flow
 
-import type { ObservableMap } from 'mobx';
-import type { EntityID } from 'brewskey.js-api';
-import type { NearbyLocation } from '../types';
+import type { Coordinates, NearbyLocation } from '../types';
+import type { IComponentStore } from './types';
 
-import { action, computed, observable, runInAction } from 'mobx';
+import { action, computed, observable } from 'mobx';
+import { LoadObject } from 'brewskey.js-api';
 import CommonApi from '../CommonApi';
 
-class NearbyLocationsStore {
-  @observable _nearbyLocationsById: ObservableMap<NearbyLocation> = new Map();
+class NearbyLocationsStore implements IComponentStore {
+  @observable
+  nearbyLocationLoader: LoadObject<Array<NearbyLocation>> = LoadObject.empty();
+  @observable _coordinates: ?Coordinates = null;
```

Comment:
This should still be a LoadObject. Otherwise things get much more complicated since you have to handle all the refreshes when a subscription changes.

Reply:
you mean I should pass `_coordinatesLoader ` from HomeScreenStore to nearbyLocastionStore, instead coordinates? 
Don't get the part about subscription changes at all.